### PR TITLE
Text File Comment Order

### PR DESCRIPTION
**Pull request by nairrrahul**
* comments on text files are now ordered by start point of the selection (#101)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -324,6 +324,35 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
...
+          //backwards selection compatibility
+          if (
+            start.line > end.line ||
+            (start.line === end.line && start.column > end.column)
+          ) {
+            [start, end] = [end, start];
+          }
+
+          for (let i = comments.length; i > 0; i--) {
+            const comment = comments.get(i - 1) as ITextSelectionComment;
```
Comment (cameron-toy) [inline src/index.ts:337]: This is out-of-bounds for i = 0. Make sure there's a case for when the end (or beginning) of the comments list has been reached.

File: src/index.ts [hunk 2]
```diff
@@ -324,6 +324,35 @@ export const jupyterCommentingPlugin: JupyterFrontEndPlugin<ICommentPanel> = {
...
+            let sel = comment.target;
+            let commentStart = sel.start;
+            //backwards selection compatibility
+            if (
```
Comment (cameron-toy) [inline src/index.ts:341]: Would a selection ever be backwards if the comment factory for text selection comments makes sure they go in in the right order?

Comment (nairrrahul) [reply src/index.ts:341]: Looks like the comment factory doesn't input them in the right order for backwards selections; I'll get onto that

File: src/index.ts (hunk 1 above)
Comment (nairrrahul) [reply src/index.ts:337]: If I'm not mistaken, why would `i` ever be 0 in the loop since the terminating condition is `> 0` not `>= 0`?

Comment (cameron-toy) [reply src/index.ts:337]: That's true. In that case , initialize index as 0 instead of comments.length

Comment (cameron-toy) [reply src/index.ts:337]: You can also just iterate through starting at 0 and reverse the comparison between the comment position and the selection position

**Comment by cameron-toy** [comment]
LGTM!

**Pull request merged**